letsencrypt/client/tests/standalone_authenticator_test.py

```diff
@@ -187,71 +187,102 @@ def setUp(self):
-        subprocess_object.communicate.return_value = ("foo", "bar")
-        subprocess_object.wait.return_value = 1
-        mock_popen.return_value = subprocess_object
-        result = self.authenticator.already_listening(17)
-        self.assertFalse(result)
-        subprocess_object.wait.assert_called_once_with()
-
-    @mock.patch("letsencrypt.client.standalone_authenticator.subprocess.Popen")
-    def test_no_relevant_line(self, mock_popen):
-        # pylint: disable=line-too-long,trailing-whitespace
-        subprocess_object = mock.MagicMock()
-        subprocess_object.communicate.return_value = (
-            """Active Internet connections (servers and established)
-Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
-tcp        0      0 127.0.1.1:53            0.0.0.0:*               LISTEN      1234/foo        
-tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      2345/bar        
-tcp        0      0 0.0.0.0:180             0.0.0.0:*               LISTEN      11111/hello     """,
-            "I am the standard error")
-        subprocess_object.wait.return_value = 0
-        mock_popen.return_value = subprocess_object
-        result = self.authenticator.already_listening(17)
-        self.assertFalse(result)
-
-    @mock.patch("letsencrypt.client.standalone_authenticator.subprocess.Popen")
+    @mock.patch("letsencrypt.client.standalone_authenticator.psutil."
+                "net_connections")
+    @mock.patch("letsencrypt.client.standalone_authenticator.psutil.Process")
     @mock.patch("letsencrypt.client.standalone_authenticator."
                 "zope.component.getUtility")
-    def test_has_relevant_line(self, mock_get_utility, mock_popen):
-        # pylint: disable=line-too-long,trailing-whitespace
-        subprocess_object = mock.MagicMock()
-        subprocess_object.communicate.return_value = (
-            """Active Internet connections (servers and established)
-Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
-tcp        0      0 127.0.1.1:53            0.0.0.0:*               LISTEN      1234/foo        
-tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      2345/bar        
-tcp        0      0 0.0.0.0:17              0.0.0.0:*               LISTEN      11111/hello     
-tcp        0      0 0.0.0.0:1728            0.0.0.0:*               LISTEN      2345/bar        """,
-            "I am the standard error")
-        subprocess_object.wait.return_value = 0
-        mock_popen.return_value = subprocess_object
+    def test_race_condition(self, mock_get_utility, mock_process, mock_net):
+        # This tests a race condition, or permission problem, or OS
+        # incompatibility in which, for some reason, no process name can be
+        # found to match the identified listening PID.
+        from psutil._common import sconn
+        from psutil import NoSuchProcess
```

Comment:
pls import at the top of the module as `import psutil` and use `psutil.NoSuchProcess instead`